llvm/lib/CodeGen/MIRPrinter.cpp

```diff
@@ -971,10 +911,11 @@ static std::string formatOperandComment(std::string Comment) {
 }
 
 void MIPrinter::print(const MachineInstr &MI, unsigned OpIdx,
-                      const TargetRegisterInfo *TRI,
-                      const TargetInstrInfo *TII,
-                      bool ShouldPrintRegisterTies, LLT TypeToPrint,
-                      bool PrintDef) {
+                      const TargetRegisterInfo *TRI, const TargetInstrInfo *TII,
+                      bool ShouldPrintRegisterTies,
+                      SmallBitVector &PrintedTypes, bool PrintDef) {
+  const auto &MRI = MI.getMF()->getRegInfo();
```

Comment:
Should pass in MRI instead of chasing through the instruction?

Reply:
Done.